tensorflow_model_optimization/python/examples/compression/keras/algorithms/weight_clustering.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
```

Comment:
I think for now, having either the tests or the examples are useful for demonstration purposes. In the long-term (as people adopt this API), it'll be good to consider their utility.

1) For the examples, the discussion at https://github.com/tensorflow/model-optimization/pull/562#issuecomment-703787341 is relevant. Hopefully adoption of these APIs will
also replace the need for us to maintain any examples ourselves (we can just point to them).

2) For the test, I'm not sure what additional behaviors are tested in this variant of weight clustering
versus same_training_and_inference_test.py.

Reply:
Okay I removed exmaples and added a test that seems only working for weight clustering. Thanks.

Reply:
> added a test that seems only working for weight clustering

I'm understanding this to be behaviors for the compression API that's unique to weight clustering and not the existing test examples. Could you clarify what aspect it is specifically?

The only thing that's obvious to me right now is that preservation of the dtype, but there is a lot more being tested than that.

Reply:
Now It tests dtype and training variable shapes, check that shapes are right shape from given parameters.
It'll be better to test the value number, but I have no idea how to do that now.